README.md

```diff
@@ -92,45 +99,129 @@ like so:
-`C:\Code\Halide`, and LLVM and Clang are checked out under
-`C:\Code\llvm`.
+We recommend building with MSVC 2019, but MSVC 2017 is also supported (earlier versions
+might work, but are not part of our tests).
+Be sure to install the CMake Individual Component in the Visual Studio 2019 installer.
+For older versions of Visual Studio, acquire CMake and Ninja from their respective project websites.
 
-    % mkdir C:\Code\llvm-build
-    % cd C:\Code\llvm-build
-    % cmake -DLLVM_ENABLE_PROJECTS="clang;lld;clang-tools-extra" -DCMAKE_INSTALL_PREFIX=../llvm-install -DLLVM_ENABLE_TERMINFO=OFF -DLLVM_TARGETS_TO_BUILD=X86;ARM;NVPTX;AArch64;Mips;Hexagon -DLLVM_ENABLE_ASSERTIONS=ON -DLLVM_BUILD_32_BITS=OFF -DCMAKE_BUILD_TYPE=Release ../llvm/llvm -G "Visual Studio 14" -A x64
+These instructions start from the `D:` drive.
+We assume this git repo is cloned to `D:\Halide`.
+We also assume that your shell environment is set up correctly. For a 64-bit build, run:
+                                                                
+    D:\> "C:\Program Files (x86)\Microsoft Visual Studio\2019\Community\VC\Auxiliary\Build\vcvarsall.bat" x64
 
-For a 32-bit build use:
+For a 32-bit build, run:
 
-    % cmake -DLLVM_ENABLE_PROJECTS="clang;lld;clang-tools-extra" -DCMAKE_INSTALL_PREFIX=../llvm-install -DLLVM_ENABLE_TERMINFO=OFF -DLLVM_TARGETS_TO_BUILD=X86;ARM;NVPTX;AArch64;Mips;Hexagon -DLLVM_ENABLE_ASSERTIONS=ON -DLLVM_BUILD_32_BITS=ON -DCMAKE_BUILD_TYPE=Release ../llvm/llvm -G "Visual Studio 14" -A Win32
+    D:\> "C:\Program Files (x86)\Microsoft Visual Studio\2019\Community\VC\Auxiliary\Build\vcvarsall.bat" x86 
 
-Then build it like so:
+#### Managing dependencies with vcpkg
+
+The best way to get compatible dependencies on Windows is to use [vcpkg](https://github.com/Microsoft/vcpkg).
+Install it like so:
+
+    D:\> git clone https://github.com/Microsoft/vcpkg.git
+    D:\> cd vcpkg
+    D:\vcpkg> .\vcpkg integrate install
+    ...
+    CMake projects should use: "-DCMAKE_TOOLCHAIN_FILE=D:/vcpkg/scripts/buildsystems/vcpkg.cmake"
+    
+Then install the libraries. For a 64-bit build, run:
+
+    D:\vcpkg> .\vcpkg install libpng:x64-windows libjpeg-turbo:x64-windows llvm[target-all,clang-tools-extra]:x64-windows
+
+To support 32-bit builds, also run:
+
+    D:\vcpkg> .\vcpkg install libpng:x86-windows libjpeg-turbo:x86-windows llvm[target-all,clang-tools-extra]:x86-windows
+
+#### Building Halide
+
+Create a separate build tree and call CMake with vcpkg's toolchain.
+This will build in either 32-bit or 64-bit depending on the environment script (`vcvars`) that was run earlier. 
+
+    D:\> md Halide-build
+    D:\> cd Halide-build
+    D:\Halide-build> cmake -G Ninja -DCMAKE_BUILD_TYPE=Release -DCMAKE_TOOLCHAIN_FILE=D:/vcpkg/scripts/buildsystems/vcpkg.cmake ..\Halide
+
+**Note:** If building with Python bindings on 32-bit (enabled by default), be sure to point CMake to the installation 
+path of a 32-bit Python3.
```

Comment:
Nit: line endings are a bit wonky. Might be nice to run all the edited .md files thru a formatter if you have one handy. If not, no worries.

Reply:
Is there one you recommend? I don't have a Markdown formatter handy.

Reply:
I just reformatted this file manually... 80 col max for prose, 120 for code.